pkg/types/types.go

```diff
@@ -216,21 +215,22 @@ type NFS struct {
 }
 
 type Ldap struct {
+	Disabled   bool   `yaml:"disabled,omitempty"`
 	Host       string `yaml:"host,omitempty"`
+	Port       string `yaml:"port,omitempty"`
 	Username   string `yaml:"username,omitempty"`
 	Password   string `yaml:"password,omitempty"`
 	Domain     string `yaml:"domain,omitempty"`
 	AdminGroup string `yaml:"adminGroup,omitempty"`
-	BindDN     string `yaml:"dn,omitempty"`
+	DN         string `yaml:"dn,omitempty"`
+	BindDN     string `yaml:"bindDN,omitempty"`
```

Comment:
how do username and BindDN differ?

Reply:
Username is used also for login in dex. It should be something like `test`.  BindDN is used for ldap first verification, something like `admin` with password `secret`.

This is what gets generated:

```yaml
- type: ldap
  id: ldap
  name: LDAP
  config:
    host: apacheds.ldap:10636
    insecureSkipVerify: true
    bindDN: uid=admin,ou=system
    bindPW: secret

    userSearch:
      baseDN: ou=users,dc=example,dc=com
      filter: "(objectClass=person)"
      username: sAMAccountName
      idAttr: DN
      emailAttr: mail
      nameAttr: displayName

    groupSearch:
      baseDN: ou=users,dc=example,dc=com
      filter: "(objectClass=group)"
      userAttr: DN
      groupAttr: member
      nameAttr: name
```

This is the config:

```yml
ldap:
  adminGroup: NA1
  username: test
  password: secret
  port: 10636
  host: apacheds.ldap
  dn: ou=users,dc=example,dc=com
  bindDN: uid=admin,ou=system
  bindPW: secret
```

Reply:
I've changed it back to defaults.